### ISSUE TITLE
getComputeStatus command exposes too much information (DBComputeJob instead of ComputeJob)

### ISSUE THREAD

**Issue by mariacarmina** [oceanprotocol/ocean-node#763]
As far as I remember for user usage, the job id with hash should be used & exposed. In fact it can be a bit misleading for end user.
 
![Screenshot 2024-11-26 at 12 34 58](https://github.com/user-attachments/assets/adb8b123-6751-4a16-a4f3-8f0a88eef784)

**Comment by paulo-ocean** [comment]
i can check this one.. thanks

**Comment by paulo-ocean** [comment]
it is used and exposed, that's why you issued the command with it :-)
the json on the response is the internal job representation (the DB data).. and internally, on the engine db,  the cluster hash is not stored..

**Comment by alexcos20** [comment]
We need to rewrite that piece of code anyway, cause now it's just dumping the internal structure, which is a security risk on one hand, and on other hand, it exposes too much information

**Comment by paulo-ocean** [comment]
> We need to rewrite that piece of code anyway, cause now it's just dumping the internal structure, which is a security risk on one hand, and on other hand, it exposes too much information

Actually, let me rephrase... its not the entire DB structure (that would be a `DBComputeJob` and way more fields)... its the partial `ComputeJob`, same thing we had on V1..
What's the sensitive data to exclude?

**Comment by paulo-ocean** [comment]
handling this one on [https://github.com/oceanprotocol/ocean-node/pull/766](https://github.com/oceanprotocol/ocean-node/pull/766)
hide internal db felds on start compute and get status responses

**Comment by paulo-ocean** [comment]
updated the issue description... when we create the compute job we always get a string in format  `hash - jobId`
That's what we use to further queries (like `getComputeStatus`, etc) even if the "internal"  `ComputeJob` format does not contain the hash.. That's not issue here. That's the same behavior as always. The issue here was that whenever we include data from the database (related with that compute job) we are revealing too much information.. we are in fact sending a `DBComputeJOb` (has more fields and more sensitive data) instead of a `computeJob`... So we need to make sure we actually do not send any of the "internal only" fields associated with DBComputeJob only

**Comment by paulo-ocean** [comment]
closed on [https://github.com/oceanprotocol/ocean-node/pull/766](https://github.com/oceanprotocol/ocean-node/pull/766)
